### PR TITLE
fix #2935: hook CPU and memory profiling back up

### LINKED ISSUE #2935: Hook CPU and memory profiling back up.

### PR DESCRIPTION
**Pull request by dgnorton**

### CODE REVIEW

File: cmd/influxd/run/server.go [hunk 1]
```diff
@@ -404,3 +417,54 @@ type Service interface {
...
 }
+
+// prof stores the file locations of active profiles.
+var prof struct {
+	cpu *os.File
+	mem *os.File
+}
+
+// StartProfile initializes the cpu and memory profile, if specified.
+func StartProfile(cpuprofile, memprofile string) {
```
Comment (otoolep) [inline cmd/influxd/run/server.go]: Minor: I don't think this needs to be exported.

File: cmd/influxd/run/server.go [hunk 2]
```diff
@@ -404,3 +417,54 @@ type Service interface {
...
+		c := make(chan os.Signal, 1)
+		signal.Notify(c, os.Interrupt)
+		<-c
+		StopProfile()
+		os.Exit(0)
+	}()
+}
+
+// StopProfile closes the cpu and memory profiles if they are running.
+func StopProfile() {
```
Comment (otoolep) [inline cmd/influxd/run/server.go]: Ditto.

**Comment by otoolep** [comment]
Seems sane, minor feedback. +1

**Comment by corylanou** [comment]
How do you use it?  I don't see how to set the memprofile or cpuprofile, so it will never start right?

**Comment by dgnorton** [comment]
@corylanou  It's a command line option:  `influxd -cpuprofile influxd.prof`

**Comment by corylanou** [comment]
Ok, I see it now. +1 Thanks!

**Pull request merged**